### PR TITLE
add finetuning lm_eval metric

### PR DESCRIPTION
**Pull request by lkk12014402**
## Type of Change
API not changed 

## Description

add evaluation metrics during finetuning.

### CODE REVIEW

File: workflows/chatbot/fine_tuning/instruction_tuning_pipeline/finetune_clm.py [hunk 1]
```diff
@@ -688,6 +668,18 @@ def concatenate_data(dataset, max_seq_length):
                     training_args.output_dir, state_dict=unwrapped_model.state_dict()
                 )
 
+        if finetune_args.do_lm_eval:
```
Comment (hshen14) [inline workflows/chatbot/fine_tuning/instruction_tuning_pipeline/finetune_clm.py:671]: how to specify the dataset e.g., TruthfulQA for Chat and CNNdaily for summerization?

Comment (kding1) [reply workflows/chatbot/fine_tuning/instruction_tuning_pipeline/finetune_clm.py:671]: in addition, pls support eval with user dataset with the metrics user wants to see (acc, acc_norm, mc2, f1 etc.)

Comment (lkk12014402) [reply workflows/chatbot/fine_tuning/instruction_tuning_pipeline/finetune_clm.py]: 1. default will use truthfulqa_mc for chat/completion evaluation, and use rouge for summarization evaluation

2. user can also use their dataset to train and eval, but need consistent with our data format or re-write preprocessing function for their dataset format.

3. different tasks use different metric, truthfulqa use mc2, lambada_openai use acc/acc_norm. user can specify the task that they want to evaluate.

Comment (hshen14) [reply workflows/chatbot/fine_tuning/instruction_tuning_pipeline/finetune_clm.py]: for 2, please add a note on how to add a custom dataset/eval in the readme.

Comment (lkk12014402) [reply workflows/chatbot/fine_tuning/instruction_tuning_pipeline/finetune_clm.py]: we have a note in README about this.
